src/main/java/com/google/sps/servlets/AccountData.java

```diff
@@ -0,0 +1,118 @@
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+
+/** Servlet that returns all account related info (links to login pages/account info). */
+@WebServlet("/account")
+public class AccountData extends HttpServlet {
+
+  public final class JsonHelper{
+        private final boolean loggedIn;
+        private final String logInOutURL;
+        private final String signUpURL;
+        public JsonHelper(Boolean loggedIn, String logInOutURL, String signUpURL){
+            this.loggedIn=loggedIn;
+            this.logInOutURL=logInOutURL;
+            this.signUpURL = signUpURL;
+          }
+        public boolean getLoggedIn(){
+            return loggedIn;
+        }
+        public String getLogInOutURL(){
+            return logInOutURL;
+        }
+        public String getSignUpURL(){
+          return signUpURL;
+        }
+    }
+
+
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    UserService userService=UserServiceFactory.getUserService();
+
+    String loginLink;
+    String signUpLink;
+    JsonHelper jsonHelper;
+
+    if (userService.isUserLoggedIn()){
+      loginLink = userService.createLogoutURL("/index.html");
+
+      Entity userEntity = getUserEntity(userService.getCurrentUser.getUserId());
+
+
+
+      jsonHelper=new JsonHelper(true, loginLink, null);
+
+
+    }
+    else{
+
+      loginLink = userService.createLoginURL("/index.html");
+      signUpLink = userService.createLoginURL("/signup.html");
+      
+
+      jsonHelper = new JsonHelper(false, loginLink, signUpLink);
+
+    }
+
+    response.setContentType("application/json");
+    response.getWriter().println(convertToJsonUsingGson(jsonHelper));
+
+
+  }
+
+
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      UserService userService = UserServiceFactory.getUserService();
+      Entity userEntity = new Entity("User");
+
+
+
+      //adds data to userentity
+      userEntity.setProperty("id",userService.getCurrentUser().getUserId());
+      userEntity.setProperty("first-name", request.getParameter("first-name"));
+      userEntity.setProperty("last-name", request.getParameter("last-name"));
+      userEntity.setProperty("email", userService.getCurrentUser().getEmail());
+
+
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.put(userEntity);
+
+      response.sendRedirect("/index.html");
+  }
+
+  private String convertToJsonUsingGson(JsonHelper jsonHelper) {
+    Gson gson = new Gson();
```

Comment:
Inconsistent indent spaces.  Should we all use two spaces as the google styling guide suggests?

Reply:
I agree with 2 spaces indent for styling.

Reply:
Indeed 2-space is the rule across all languages used at Google, I believe.

For reviewers: When you suggest a style change, it's good to reference the style guide: https://google.github.io/styleguide/javaguide.html#s4.2-block-indentation